### PR TITLE
[Explorer] Data selector enhancement and refactoring adoptions

### PR DESCRIPTION
**Pull request by mengweieric**
### Description
Data selector in core has a set of enhancements and refactoring which require the same counterpart in explorer to also change accordingly.

https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6571]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Swiddis — APPROVED** [review]

File: common/constants/shared.ts [hunk 1]
```diff
@@ -258,7 +258,7 @@ export const VISUALIZATION_ERROR = {
   NO_METRIC: 'Invalid Metric MetaData',
 };
 
-export const S3_DATASOURCE_TYPE = 'S3_DATASOURCE';
+export const S3_DATA_SOURCE_TYPE = 's3glue';
```
Comment (YANG-DB) [inline common/constants/shared.ts:261]: why s3glue and not glue ?

File: public/framework/datasources/obs_opensearch_datasource.ts [hunk 2]
```diff
@@ -6,25 +6,26 @@
 import { DataSource } from '../../../../../src/plugins/data/public';
 
 interface DataSourceConfig {
+  id: string;
```
Comment (YANG-DB) [inline public/framework/datasources/obs_opensearch_datasource.ts:9]: can we add a list of labels (string array) for future filtering and association support ?

File: public/plugin.tsx [hunk 3]
```diff
@@ -387,38 +394,58 @@ export class ObservabilityPlugin
...
           dataSourceService.registerDataSource(
-            dataSourceFactory.getDataSourceInstance(S3_DATASOURCE_TYPE, {
+            dataSourceFactory.getDataSourceInstance(S3_DATA_SOURCE_TYPE, {
+              id: htmlIdGenerator(OBS_S3_DATA_SOURCE)(DATA_SOURCE_TYPES.S3Glue),
               name: s3ds.name,
               type: s3ds.connector.toLowerCase(),
-              metadata: s3ds,
+              metadata: {
+                ...s3ds.properties,
+                ...{
```
Comment (joshuali925) [inline public/plugin.tsx:410]: why is it spreading an object?

File: common/constants/data_sources.ts [hunk 4]
```diff
@@ -90,3 +90,7 @@ export const ACCELERATION_AGGREGRATION_FUNCTIONS = [
 ];
 
 export const SPARK_PARTITION_INFO = `# Partition Information`;
+export const OBS_DEFAULT_CLUSTER = 'observability-default';
```
Comment (joshuali925) [inline common/constants/data_sources.ts:93]: maybe some comments on these? e.g. what is an observability default cluster

**Review by joshuali925 — APPROVED** [review]

**Comment by Swiddis** [comment]
It looks like all of the integration test actions are failing, including the ones that weren't failing before -- Can you investigate?

File: public/plugin.tsx (hunk 3 above)
Comment (mengweieric) [reply public/plugin.tsx:410]: include any other metadata in s3 properties

File: common/constants/shared.ts (hunk 1 above)
Comment (mengweieric) [reply common/constants/shared.ts:261]: this is for data selector grouping, the s3 data source type returned from backend is s3glue

File: public/plugin.tsx (hunk 3 above)
Comment (joshuali925) [reply public/plugin.tsx:410]: comment is for line 410, not 409

**Pull request merged**